ocpp/charge_point.py

```diff
@@ -165,17 +167,32 @@ async def _handle_call(self, msg):
 
         First the '_on_action' hook is executed and its response is returned to
         the client. If there is no '_on_action' hook for Action in the message
-        a CallError with a NotSupportedError is returned.
+        a CallError with a NotImplementedError is returned. If the Action is
+        not supported by the OCPP version a NotSupportedError is returned.
 
         Next the '_after_action' hook is executed.
 
         """
         try:
             handlers = self.route_map[msg.action]
         except KeyError:
-            raise NotSupportedError(
-                details={"cause": f"No handler for {msg.action} registered."}
-            )
+            if self._ocpp_version == "1.6":
+                if hasattr(v16_Action, msg.action):
+                    raise NotImplementedError(
+                        details={"cause": f"No handler for {msg.action} registered."}
+                    )
+            elif self._ocpp_version in ["2.0", "2.0.1"]:
+                if hasattr(v201_Action, msg.action):
+                    raise NotImplementedError(
+                        details={"cause": f"No handler for {msg.action} registered."}
+                    )
+            else:
+                raise NotSupportedError(
+                    details={
+                        "cause": f"{msg.action} not supported by OCPP"
+                        "{self._ocpp_version}."
+                    }
+                )
```

Comment:
This code is repeated twice. I suggest to isolate it to a function. 

Since the logic doesn't really need access to self, I don't think a method bound to the `ChargePoint` class is needed. Instead, just create a "private" function in the module scope.

Reply:
Thanks, done